### PR TITLE
Fixes #1745 - Use SlaveID for reconciliation

### LINKED ISSUE #1745: Save Task SlaveID and use it for reconciliation
This will speedup reconciliation in the presence of disconnected slaves. See API section in 

http://mesos.apache.org/documentation/latest/reconciliation/

This might improve #1279 and #1456.

Of course, the code has to work in the case the SlaveID was not yet saved for a task.

### PR DESCRIPTION
**Pull request by janisz**

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 1]
```diff
@@ -449,6 +449,12 @@ class SchedulerActions(
         val knownTaskStatuses = appIds.flatMap { appId =>
           taskTracker.get(appId).collect {
             case task: Protos.MarathonTask if task.hasStatus => task.getStatus
+            case task: Protos.MarathonTask if task.hasSlaveId => // staged tasks without status but with slaveId
```
Comment (kolloch) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Great patch, @janisz!

I would prefer if stored the builder in a var:

`val builder = TaskStatus.newBuilder` ...
and had an `if (task.hasSlaveId)` that adds the slaveId. That makes it more obvious that this case only differs in the slaveId. Now we have two nearly identical code blocks...

**Comment by kolloch** [comment]
Great patch, thanks! Had I minor comment. Does it make sense to you?

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala (hunk 1 above)
Comment (janisz) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: How about extracting common code to method?

**Comment by janisz** [comment]
I extracted duplicated code, it looks better.

**Comment by kolloch** [comment]
Thanks, @janisz !

**Pull request merged**